### ISSUE TITLE
(C++) Errors when parsing friend functions in a class

### ISSUE THREAD

**Issue by Grace-Li-cam** [michaeljones/breathe#613]
Hi, I have got this error:

In a class I defined some friend functions:
```

class D {

public:

friend D operator+(const D &x, const D &y);

friend D operator+(const int64_t &x, const D &y);
};
```

Doxygen generated XML correctly.

![image](https://user-images.githubusercontent.com/76575256/103114342-21721f80-4699-11eb-8ba7-dd8d9efc90f9.png)

In the corresponding rst:
```

operator+
>>>>>>>>>>>>>

.. doxygenfunction:: mm::D::operator+(const D &x, const D &y)

operator+
>>>>>>>>>>>>>

.. doxygenfunction:: mm::D::operator+(const int64_t &x, const D &y)
```

But when running ./makelatexpdf.sh to generate PDF, the follow information is returned:

WARNING: doxygenfunction: Cannot find function "mm::D::operator+" in doxygen xml output for project ...

Breathe version: 4.24.1
Sphinx version: 3.3.1

**Comment by awsteiner** [comment]
If I may, what version of doxygen are you using? I'm having some friend problems with doxygen 1.9 and I'm curious if you're having a similar problem.

**Comment by Grace-Li-cam** [comment]
I am using doxygen 1.8.18. What doxygen version would you recommend to use?

**Comment by awsteiner** [comment]
I have no recommendation myself, I was just trying to see if anyone had the same issue I was having, but now I think that my issue is the same as #616 and thus different than this issue.

**Comment by Grace-Li-cam** [comment]
The friend functions can be parsed correctly if using ".. doxygenindex::". But all the functions cannot be included in the table of contents in the generated PDF. Is there a way to include them in the table of contents?

**Comment by Grace-Li-cam** [comment]
> If I may, what version of doxygen are you using? I'm having some friend problems with doxygen 1.9 and I'm curious if you're having a similar problem.

Doxygen generates the XML correctly. I am wondering if the problem is from Doxygen?

**Comment by vermeeren** [comment]
@Grace-Li-cam It should work with Breathe, but I remember the syntax for specifying the signature can be odd sometimes requiring perfect input or it may fail to find the function.

Can you try:

```rst
.. doxygenfunction:: mm::D::operator+
```

This should complain about it being ambiguous, showing the available outputs in the warning/error output I believe. That should reveal the exact syntax needed. This is all from memory, so I could be wrong.

**Comment by vermeeren** [comment]
Just released v4.26.1 containing the fix.
